### PR TITLE
Added Content Response on Write Enabled feature to CosmosItemRequestOptions

### PR DESCRIPTION
**Pull request by kushagraThapar**
* Added `contentResponseOnWriteEnabled` field to `CosmosItemRequestOptions`
* This field if set `true` or `false`, will override whatever is set in CosmosClientBuilder for this particular request on to which these options are passed. 
* By default, value of this field is `null`, because to support cases where customer has set this field in `CosmosClientBuilder` as `true` and wants to enable or disable the content response for an individual requests. 
* Improved testing scenarios with all combinations for this field in `CosmosClientBuilder` and `CosmosItemRequestOptions`
* Feature request - https://github.com/Azure/azure-sdk-for-java/issues/16076
* Closes - https://github.com/Azure/azure-sdk-for-java/issues/16076

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosItemRequestOptions.java [hunk 1]
```diff
@@ -203,6 +204,51 @@ public CosmosItemRequestOptions setSessionToken(String sessionToken) {
...
+     */
+    public Boolean isContentResponseOnWriteEnabled() {
+        return contentResponseOnWriteEnabled;
+    }
+
+    /**
+     * Sets the boolean to only return the headers and status code in Cosmos DB response
+     * in case of Create, Update and Delete operations on CosmosItem.
+     *
+     * If set to false, this removes the resource from response. It reduces networking
```
Comment (moderakh) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosItemRequestOptions.java:232]: two comments:
1. previously we had the notion of document/resource. Now there is not resource notion anywhere in our api or javadoc.
I would change "resource" to payload or data item.
2. "this removes the resource from response". Instead of saying "remove", I would reword this to say service doesn't return the payload. as it may become confusing what "removing" means.

same for the setter api.

**Review by moderakh — APPROVED** [review]
one minor comment on the javaapi doc.
other than that LGTM. awesome feature. Thanks.

Also we should track changing CFP to rely on this new awesome request option rather than forcing the ContentResponseOnWrite be enabled globally (this is outside of the scope of the PR, please just track it.)

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosItemRequestOptions.java (hunk 1 above)
Comment (kushagraThapar) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosItemRequestOptions.java:232]: Good points, will make these changes across javadocs and will also check for CFP.

Comment (kushagraThapar) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosItemRequestOptions.java]: @moderakh - updated the java docs.
